charts/linkerd-control-plane/templates/proxy-injector.yaml

```diff
@@ -118,7 +118,7 @@ spec:
           name: tls
           readOnly: true
       initContainers:
-      {{ if .Values.cniEnabled -}}
+      {{ if and (.Values.cniEnabled) (.Values.validatorEnabled) -}}
```

Comment:
I think the way this `if-else` is structured, setting `validatorEnabled` will cause the `else` branch to be used which adds the proxy-init container.  If `cniEnabled` is true, but `validatorEnabled` is false, I think we actually want neither of the proxy-init or the networkValidator, right?